jelares/portfolio/src/main/webapp/javascript/blog.js

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** Fetches the URL for uploading to Blobstore and adds it to the blog form */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url')
+      .then((response) => {
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('blog-post');
+        messageForm.action = imageUploadUrl;
+      });
+}
+
+/** Adds the blog posts with content, titles, and images, to the blog-posts element */
+function getPosts() {
+  console.log("fetching posts");
+  const numPostsElement = document.getElementById('num-posts');
+  const numPosts = numPostsElement.value;
+
+  fetch(`/blog-data?num-posts=${numPosts}`)
+      .then(response => response.json())
+      .then((posts) => {
+        const postListElement = document.getElementById('blog-posts');
+        postListElement.innerHTML = '';
+        const postObjectArray = Object.values(posts)[0];
+
+        for (let i = 0; i < postObjectArray.length; i++) {
+          const content = postObjectArray[i].content;
+          const title = postObjectArray[i].title;
+          const imageUrl = postObjectArray[i].imageUrl;
+
+          postListElement.appendChild(
+              createListElement(content, title, imageUrl));
+        }
+      });
+}
+
+/** @return listelement, an <li> blog post element containing the blog post content, title, and image */
```

Comment:
JSDoc has a format for specifying return type, which might help here:
https://google.github.io/styleguide/jsguide.html#jsdoc

```
@return {Element} An <li> element containing ...
```

Reply:
Okay thanks for the heads up, will use that from now on.